WcaOnRails/app/models/post.rb

```diff
@@ -2,11 +2,37 @@
 
 class Post < ApplicationRecord
   belongs_to :author, class_name: "User"
+  has_many :post_tags, autosave: true, dependent: :destroy
 
   validates :title, presence: true, uniqueness: true
   validates :body, presence: true
   validates :slug, presence: true, uniqueness: true
 
+  def reload
+    @tags = nil
+    super
+  end
+
+  attr_writer :tags
+
+  def tags
+    @tags ||= post_tags.pluck(:tag).join(",")
```

Comment:
The user is giving us a string, *somewhere* we need to split it on commas to turn it into an array of tags. There are probably a few ways to accomplish this. As you've guessed, the current implementation works through an interaction between `attr_writer :tags` (which writes to `@tags`) and `def tags` (which reads from `@tags` if set, otherwise it goes to the database).